src/Serilog.AspNetCore/AspNetCore/RequestLoggingMiddleware.cs

```diff
@@ -71,10 +72,10 @@ public async Task Invoke(HttpContext httpContext)
         bool LogCompletion(HttpContext httpContext, DiagnosticContextCollector collector, int statusCode, double elapsedMs, Exception ex)
         {
             var logger = Log.ForContext<RequestLoggingMiddleware>();
-            var level = statusCode > 499 ? LogEventLevel.Error : LogEventLevel.Information;
+            var level = _getLevel(httpContext);
```

Comment:
I just realized that we've introduced a regression in the events raised via:

```
                when (LogCompletion(httpContext, collector, 500, GetElapsedMilliseconds(start, Stopwatch.GetTimestamp()), ex))
```

In this case, the 500 is being passed explicitly because it won't yet have been set on `HttpContext.Response.StatusCode`. There's a bit of awkwardness around this and we do sometimes lie about the status code because of this.

We could resolve this by making the signature of `_getLevel` accept both the context and the exception, and treat the level as 500 in `DefaultGetLevel` if the exception is present.

Reply:
I had a similar problem in the past. An exception is raised but StatusCode is not yet set to 500. I changed the `Func` to accept the `HttpContext` and the `Exception`.  I'm tempted to create a custom context class exposing fewer properties. This context could be built before calling `_getLevel`. This class would contain a `HttpStatusCode` (500 if exception), the `Exception` (if any) and the Request full URL. It gives less flexibility than passing the HttpContext but at least we can control the values we pass to the Func. What do you think about that?

Reply:
I think `HttpContext` is a stable enough type that we can probably expose it as-is; otherwise we'll have to do a lot of guesswork to support the right scenarios.

Thanks for making those last few changes! I'll loop back to this and get it merged shortly.